bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -0,0 +1,537 @@
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.thing.type.ChannelKind;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.StateOption;
+import org.openhab.binding.intesis.internal.IntesisConfiguration;
+import org.openhab.binding.intesis.internal.IntesisDynamicStateDescriptionProvider;
+import org.openhab.binding.intesis.internal.IntesisHomeModeEnum;
+import org.openhab.binding.intesis.internal.api.IntesisHomeHttpApi;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.Data;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.Datapoints;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.Descr;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.Dp;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.Dpval;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.Id;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.Info;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link IntesisHomeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Hans-Jörg Merk - Initial contribution
+ */
+@NonNullByDefault
+public class IntesisHomeHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(IntesisHomeHandler.class);
+    private final IntesisHomeHttpApi api;
+    private IntesisConfiguration config = new IntesisConfiguration();
+
+    private final Map<String, String> properties = new HashMap<>();
+
+    private IntesisDynamicStateDescriptionProvider intesisStateDescriptionProvider;
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    private String ipAddress = "";
+    private String password = "";
+
+    final Gson gson = new Gson();
+
+    public IntesisHomeHandler(final Thing thing, final HttpClient httpClient,
+            IntesisDynamicStateDescriptionProvider intesisStateDescriptionProvider) {
+        super(thing);
+        this.api = new IntesisHomeHttpApi(config, httpClient);
+        this.intesisStateDescriptionProvider = intesisStateDescriptionProvider;
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        final IntesisConfiguration config = getConfigAs(IntesisConfiguration.class);
```

Comment:
I assume it was your original intention save the config in your `config` field. This is the standard design pattern in most bindings.
```suggestion
        config = getConfigAs(IntesisConfiguration.class);
```

Reply:
Done